azurerm/internal/services/customproviders/tests/resource_arm_custom_resource_provider_test.go

```diff
@@ -0,0 +1,252 @@
+	})
+}
+
+func TestAccAzureRMCustomResourceProvider_action(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_custom_resource_provider", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMCustomResourceProviderDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMCustomResourceProvider_action(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMCustomResourceProviderExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMCustomResourceProvider_actionUpdate(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMCustomResourceProviderExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMCustomResourceProvider_action(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMCustomResourceProviderExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func testCheckAzureRMCustomResourceProviderExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).CustomProviders.CustomResourceProviderClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		id, err := parse.CustomResourceProviderID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+		if err != nil {
+			return fmt.Errorf("Bad: Get on CustomResourceProviderClient: %+v", err)
+		}
+
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Bad: Custom Resource Provider %q (resource group: %q) does not exist", id.Name, id.ResourceGroup)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMCustomResourceProviderDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).CustomProviders.CustomResourceProviderClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_custom_resource_provider" {
+			continue
+		}
+
+		id, err := parse.CustomResourceProviderID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("Custom Resource Provider still exists: %q", id.Name)
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureRMCustomResourceProvider_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-cr-%d"
```

Comment:
should this be
```suggestion
  name     = "acctestRG-cp-%d"
```
for custom providers?